### DISCUSSION TITLE
Why does Tauri 2.0 make the Invoke struct non-pub?

### DISCUSSION THREAD

**Discussion by suxiaoshao** [tauri-apps/tauri#11250] [Q&A] [answered]
In the Tauri 1.x version, developers can implement the `tauri::plugin::Plugin` trait for custom structures. The `extend_api `method requires using the `Invoke` structure, which is currently not exported. Does this mean that developers can no longer implement their own `Plugin` and must use `Plugin::Builder` to create a plugin?

**Comment by FabianLars** [comment]
This sounds like an accident. Please open a normal GitHub issue so that the team can take a look.

P.S. yes, the Builder is basically always recommended. Would you mind explaining your issues with it, if any?

**Reply by suxiaoshao** [reply, marked as answer]
I found the problem; the `Invoke` in the Tauri code has `#[doc(hidden)]` added, which is why I couldn't find it on docs.rs and rust-analyzer didn't auto-import it either.

**Reply by suxiaoshao** [reply]
Regarding the `Builder`, there are times when the invoke function may need to change based on the data from the `Plugin`, so at that moment, the `Builder` does not meet my needs.
